src/policy/immix/block.rs

```diff
@@ -201,6 +201,9 @@ impl Block {
                     #[cfg(feature = "vo_bit")]
                     vo_bit::helper::on_region_swept::<VM, _>(self, false);
 
+                    #[cfg(feature = "object_pinning")]
```

Comment:
This should be under the condition that the pinning bit metadata is on the side.  If the pinning bit is in the header, we don't need to clear them because the object will be overwritten anyway.

Reply:
I'm doing that check to only bulk clear if it's on the side in the `bzero_pin_bit` function (see [here](https://github.com/udesou/mmtk-core/blob/98f5a67ff7017649ba55302a0ac8a0a6c7615802/src/util/metadata/pin_bit.rs#L55-L63)), is your suggestion to instead "inline" that check when sweeping the block?

Reply:
> I'm doing that check to only bulk clear if it's on the side in the `bzero_pin_bit` function (see [here](https://github.com/udesou/mmtk-core/blob/98f5a67ff7017649ba55302a0ac8a0a6c7615802/src/util/metadata/pin_bit.rs#L55-L63)), is your suggestion to instead "inline" that check when sweeping the block?

I think that is what Kunshan meant. If the pin bits is not on the side, there is no need to call bulk zero at all, as we cannot and we dont need to bulk zero the bits.